src/main/java/org/openstreetmap/atlas/geography/atlas/Atlas.java

```diff
@@ -688,6 +691,32 @@
      */
     void saveAsGeoJson(WritableResource resource, Predicate<AtlasEntity> matcher);
 
+    /**
+     * Save as line-delimited GeoJSON. This is one feature per line, with no wrapping
+     * FeatureCollection.
+     *
+     * @param resource
+     *            The resource to write to
+     * @param jsonMutator
+     *            The callback function that will let you change what is in the Feature's JSON.
+     */
+    void saveAsLineDelimitedGeoJsonFeatures(WritableResource resource,
+            BiConsumer<AtlasEntity, JsonObject> jsonMutator);
+
+    /**
+     * Save as line-delimited GeoJSON with a matcher. This is one feature per line, with no wrapping
+     * FeatureCollection.
+     *
+     * @param resource
+     *            The resource to write to
+     * @param matcher
+     *            The matcher to consider
+     * @param jsonMutator
+     *            The callback function that will let you change what is in the Feature's JSON.
+     */
+    void saveAsLineDelimitedGeoJsonFeatures(WritableResource resource,
+            Predicate<AtlasEntity> matcher, BiConsumer<AtlasEntity, JsonObject> jsonMutator);
```

Comment:
I disagree on this one. the `JsonObject jsonMutator` object is being mutated, as the name suggests. We pass this object by reference, and we do things to it. This is a very common pattern, both in Java and C++. If you were to also return it in a function, that would imply that you're copying or making a new object based on the input parameters, which we are not. Doing that would be a poor choice, as it would hurt performance and do an unnecessary copy on something that is being used exclusively for this given purpose.

Reply:
The BiFunction wouldn't necessitate a new object be made. This isn't a blocker for me but the pattern does make more compact implementations possible and fits with the Java8 stream usage patterns.

Reply:
I agree with @hallahan here. From the documentation of `BiConsumer`:
```java
/**
 * Represents an operation that accepts two input arguments and returns no
 * result.  This is the two-arity specialization of {@link Consumer}.
 * Unlike most other functional interfaces, {@code BiConsumer} is expected
 * to operate via side-effects.
.
.
.
@FunctionalInterface
public interface BiConsumer<T, U> {
.
.
.
```
When you see a `Consumer`, you know that you're dealing with mutation via pass-by- reference semantics - side effects are a well-documented (and expected) behavior. Since `Atlas` and its entities are immutable, it's implied here that the `JsonObject` is the subject of mutation.